portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -29,17 +37,27 @@
 @WebServlet("/comment")
 public class DataServlet extends HttpServlet {
 
-  private ArrayList<String> comments = new ArrayList<String>();
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int numComments = Integer.parseInt(request.getParameter("num"));
```

Comment:
`Integer.parseInt` will throw a `NumberFormatException` if you pass a bad value. In this case, I think you would end up with a `500 Internal Server Error`. Instead, it would be better to catch it and respond with a `400 Bad Request`.